src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/ChecklistServiceImpl.java

```diff
@@ -59,10 +66,202 @@ public ChecklistServiceImpl(ApplicationData applicationData, Tools tools, InforW
+		return getResult;
+	}
+
+	public WorkOrderActivityChecklistSignatureResponse eSignWorkOrderActivityChecklist(InforContext context, WorkOrderActivityCheckListSignature workOrderActivityCheckListSignature)
+			throws Exception {
+		Signature signature = new Signature();
+		signature.setUserCode(workOrderActivityCheckListSignature.getUserCode());
+		signature.setPassword(workOrderActivityCheckListSignature.getPassword());
+		signature.setSignatureType(workOrderActivityCheckListSignature.getSignatureType());
+		context.setSignature(signature);
+		if(signature.getSignatureType().endsWith("02"))
+			workOrderActivityCheckListSignature.setSequenceNumber(new BigInteger("2"));
+		if(workOrderActivityCheckListSignature.getSignatureType().startsWith("PB")) {
+			workOrderActivityCheckListSignature.setVerb(VERB_Type.PERFORM);
+			return  performWorkOrderActivityChecklist(context, workOrderActivityCheckListSignature);
+		} else {
+			workOrderActivityCheckListSignature.setVerb(VERB_Type.REVIEW);
+			return reviewWorkOrderActivityCheckList(context, workOrderActivityCheckListSignature);
+		}
+	}
+
+	private WorkOrderActivityChecklistSignatureResponse performWorkOrderActivityChecklist(InforContext context, WorkOrderActivityCheckListSignature workOrderActivityCheckListSignature)
+			throws Exception {
+		DateAdapter dateAdapter = new DateAdapter();
+		MP7997_PerformWorkOrderActivityCheckList_001 performWorkOrderActivityCheckList = new MP7997_PerformWorkOrderActivityCheckList_001();
+
+		tools.getInforFieldTools().transformWSHubObject(performWorkOrderActivityCheckList, workOrderActivityCheckListSignature, context);
+		tools.performInforOperation(context, inforws::performWorkOrderActivityCheckListOp, performWorkOrderActivityCheckList);
+
+		MP7999_GetWorkOrderActivityCheckListDefault_001_Result getResult =
+				getSignatureWS(context, workOrderActivityCheckListSignature.getWorkOrderCode(),
+						       workOrderActivityCheckListSignature.getActivityCodeValue().toString());
+		WorkOrderActivityChecklistSignatureResponse res = new WorkOrderActivityChecklistSignatureResponse();
+		if(workOrderActivityCheckListSignature.getSequenceNumber() != null &&
+				workOrderActivityCheckListSignature.getSequenceNumber().intValue() == 2){
+			res.setSigner(getResult.getResultData().getWorkOrderActivityCheckListDefault().getPERFORMEDBYESIGN2().getESIGNATURE().getUSERID().getDESCRIPTION());
+			res.setTimeStamp(dateAdapter.marshal(decodeInforDate(
+					getResult.getResultData().getWorkOrderActivityCheckListDefault().getPERFORMEDBYESIGN2().getESIGNATURE().getEXTERNALDATETIME())));
+		}
+		else{
+			res.setSigner(getResult.getResultData().getWorkOrderActivityCheckListDefault().getPERFORMEDBYESIGN().getESIGNATURE().getUSERID().getDESCRIPTION());
+			res.setTimeStamp(dateAdapter.marshal(decodeInforDate(
+					getResult.getResultData().getWorkOrderActivityCheckListDefault().getPERFORMEDBYESIGN().getESIGNATURE().getEXTERNALDATETIME())));
+		}
```

Comment:
Is this not possible to do with @InforField annotations?

Reply:
The response of the webservice contains information for all three E-Signature types. Thus to which fields we need to bind the Response fields cannot be determined at compile time. A workaround will be to change the Response class to contain fields for all three signature types. That would mean, however, that the front end will have to determine which part of the response to display. Is there another approach to annotating the response class and if not, is adding more work to the front end worth it?